packages/ui/src/app/pages/Validators/ValidatorList.stories.tsx

```diff
@@ -113,4 +143,78 @@ export default {
+    const stateFilter = screen.getAllByText('State')[0]
+    const clearFilter = async () => {
+      expect(screen.getByText('Clear all filters'))
+      await userEvent.click(screen.getByText('Clear all filters'))
+    }
+    const expectNumberOfValidatorsToBe = async (n: number) => {
+      const liElements = screen.queryAllByRole('listitem')
+      const validatorElements = liElements.filter((li) => li.textContent?.includes('Nominate'))
+      await expect(validatorElements.length).toBe(n)
+    }
+
+    await step('Verifcation Filter', async () => {
+      await selectFromDropdown(screen, verificationFilter, 'verified')
+      await expect(screen.queryByText('unverifed')).toBeNull()
+      await expectNumberOfValidatorsToBe(4)
+      await expect(screen.getByText('alice'))
+      await expect(screen.getByText('bob'))
+      await selectFromDropdown(screen, verificationFilter, 'unverified')
+      await expect(screen.queryByText('verifed')).toBeNull()
+      await expectNumberOfValidatorsToBe(7)
+      await expect(screen.queryByText('alice')).toBeNull()
+      await expect(screen.queryByText('bob')).toBeNull()
+      await selectFromDropdown(screen, verificationFilter, 'All')
+    })
+    await step('State Filter', async () => {
+      await selectFromDropdown(screen, stateFilter, 'active')
+      await expect(screen.queryByText('waiting')).toBeNull()
+      await expectNumberOfValidatorsToBe(9)
+      await selectFromDropdown(screen, stateFilter, 'waiting')
+      await expect(screen.queryByText('active')).toBeNull()
+      await expectNumberOfValidatorsToBe(2)
+      await selectFromDropdown(screen, stateFilter, 'All')
+    })
+    await step('Search', async () => {
+      await userEvent.type(searchElement, 'j4Rh1c')
+      await waitFor(async () => {
+        await userEvent.type(searchElement, '{enter}')
+        expect(screen.queryAllByRole('button', { name: 'Nominate' })).toHaveLength(1)
+      })
+      await expect(screen.getByText('alice'))
+      await userEvent.clear(searchElement)
+      await userEvent.type(searchElement, 'j4R')
+      await waitFor(async () => {
+        await userEvent.type(searchElement, '{enter}')
+        expect(screen.queryAllByRole('button', { name: 'Nominate' })).toHaveLength(9)
+      })
+      await expect(screen.getByText('alice'))
+      await expect(screen.getByText('bob'))
+    })
+
+    await step('Clear Filter', async () => {
+      await selectFromDropdown(screen, verificationFilter, 'verified')
+      await expect(screen.getByText('Clear all filters'))
+      await selectFromDropdown(screen, stateFilter, 'waiting')
+      await expectNumberOfValidatorsToBe(0)
+      await selectFromDropdown(screen, stateFilter, 'active')
+      await expectNumberOfValidatorsToBe(4)
+      await clearFilter()
+      await expectNumberOfValidatorsToBe(11)
+      await userEvent.type(searchElement, 'j4R{enter}')
+      await expectNumberOfValidatorsToBe(9)
```

Comment:
I have no idea why this one appear to work but to be safe I think it should be changed too:
```suggestion
      await userEvent.type(searchElement, 'j4R')
      await waitFor(async () => {
        await userEvent.type(searchElement, '{enter}')
        expect(screen.queryAllByRole('button', { name: 'Nominate' })).toHaveLength(9)
      })
```

Reply:
thanks